pkg/controller/machine/machine_controller.go

```diff
@@ -1217,7 +1217,7 @@ ExecStart=/opt/bin/bootstrap
 hostname: {{ .MachineSpec.Name }}
 {{- /* Never set the hostname on AWS nodes. Kubernetes(kube-proxy) requires the hostname to be the private dns name */}}
 {{ end }}
-ssh_pwauth: no
+ssh_pwauth: false
```

Comment:
Why is this changing to `false`? Maybe I'm looking at the wrong documentation, but https://cloudinit.readthedocs.io/en/latest/topics/modules.html#set-passwords suggests that `no` is the correct value to set.

Reply:
This happens in a couple of places, but I'll leave this as one review comment to discuss this.

Reply:
After going over the code with @xrstf it looks like cloud-init is capable of understanding variations of true/false here, including yes/no, which is what is referenced in the docs for some reason. But this should be fine as well.

Reply:
https://github.com/number5/cloud-init/blob/main/cloudinit/config/cc_set_passwords.py#L101 shows that the Python code expects a true and https://github.com/number5/cloud-init/blob/820e3d1c1c8869aa84757bcd49b8619c17a350b1/cloudinit/util.py#L219 proves that a native bool is even faster (FASTER!!!11!!), so I think we're find. cloudinit is also translating this into the appropriate value, regardless whether it was "no" or "false".

I just got reminded the of YAML alias problem and wanted to avoid one more pitfall ^^